students/navdeep/session04/file_lab.py

```diff
@@ -0,0 +1,58 @@
+import os
+
+def copyFile(oldFile):
+    """
+    Will copy the contents of an old file to a new file.  Function can read binary
+    or text file.
+    :param oldFile: the original file we want to copy contents from
+    """
+    newfile = input("Enter name of new file (include file type): ")
+    with open(newfile, 'wb+') as output:
+        while True:
+            read_data = oldFile.read(1000)
+            print(type(read_data))
+            if str(read_data) == "b\'\'":
+                break
+            output.write(read_data)
+    output.close()
```

Comment:
The with open statement is pretty cool, because when you leave that block, it automatically shuts the file for you, so the output.close() is not necessary. You probably do want to close the oldFile, however, since it isn't necessarily clear where it is coming from or if it will be closed when the copyFile function exits.